zio-http/src/main/scala/zio/http/codec/internal/EncoderDecoder.scala

```diff
@@ -38,11 +38,14 @@ private[codec] trait EncoderDecoder[-AtomTypes, Value] {
   def encodeWith[Z](value: Value)(f: (URL, Option[Status], Option[Method], Headers, Body) => Z): Z
 }
 private[codec] object EncoderDecoder                   {
-  def apply[AtomTypes, Value](httpCodec: HttpCodec[AtomTypes, Value]): EncoderDecoder[AtomTypes, Value] = {
+  def apply[AtomTypes, Value](httpCodec: HttpCodec[AtomTypes, Value]): Option[EncoderDecoder[AtomTypes, Value]] = {
     val flattened = httpCodec.alternatives
 
-    if (flattened.length == 1) Single(flattened.head)
-    else Multiple(flattened)
+    flattened.length match {
+      case 0 => None
```

Comment:
Instead of this we could introduce a third `EncoderDecoder`, let's say `Undefined`, with its encode and decode methods just failing. Then you don't have to modify `HttpCodec`s `decode` and `encodeWith` either.